src/scheduler/pbs_sched.cpp

```diff
@@ -582,20 +586,23 @@ connect_svrpool()
 			 * Also wait for 2s for not to burn too much CPU
 			 */
 			sleep(2);
+			close_servers();
 			continue;
 		}
 
-		if (pbs_register_sched(sc_name) == 0) {
+		if (pbs_register_sched(sc_name, clust_primary_sock, clust_secondary_sock) == 0) {
 			log_errf(pbs_errno, __func__, "Couldn't register the scheduler %s with the configured servers", sc_name);
 			/* wait for 2s for not to burn too much CPU, and then retry connection */
-			sleep(2);			
+			sleep(2);
+			close_servers();			
```

Comment:
you will need to set clust_primary_sock and clust_secondary_sock to -1 every time you call close_servers() so that you can connect with them again. I think you should just have a while(1) loop instead of while (clust_primary_sock < 0 || clust_secondary_sock < 0 || registered == 0). Then, you don't need clust_primary_sock, clust_secondary_sock or registered, and you can just break out of the loop at the end when you have connected and registered both primary and secondary.

Reply:
> you will need to set clust_primary_sock and clust_secondary_sock to -1 every time you call close_servers() so that you can connect with them again

This is not required since I am setting set clust_primary_sock and clust_secondary_sock in close_servers() itself.

Reply:
> I think you should just have a while(1) loop instead of while (clust_primary_sock < 0 || clust_secondary_sock < 0 || registered == 0). Then, you don't need clust_primary_sock, clust_secondary_sock or registered, and you can just break out of the loop at the end when you have connected and registered both primary and secondary.

I have kept all the conditions in while loop just for better readability. Now I have changed it to while(1) loop and breaking it if everything is succeeded. Please check my latest commit.